### PR TITLE
RM14: Structured types (list, struct and map)

### PR DESCRIPTION
**Pull request by jghoman**
## Summary

- Graduate RM14: LIST, STRUCT, and MAP nested types fully working over Arrow Flight
- Add `ValueToInsertSQL` serializer for STRUCT/MAP/LIST with proper SQL quoting
- Fix single-quote escaping in STRUCT field names (latent injection bug)
- Comprehensive test coverage across all layers: C++ unit, DuckLake conformance, Flight integration

## What changed

**Serializer** (`posthog_sql_utils.cpp`):
- `ValueToInsertSQL` now handles MAP (`MAP {'k': v}`), STRUCT (`{'field': v}`), and LIST (`[v]`) recursively
- STRUCT field names are single-quote escaped via `StringUtil::Replace`
- DuckDB's `Value::ToSQLString()` falls through to `ToString()` for MAP, producing `{k=v}` (display format, not valid SQL) — our serializer emits `MAP {'k': v}` instead

**Integration tests** (`struct_types_remote.test_slow`, `map_types_remote.test_slow`):
- Full CRUD round-trip over Flight for STRUCT and MAP types
- Nested combinations: STRUCT-in-STRUCT, LIST-in-STRUCT, STRUCT-in-LIST, MAP-in-STRUCT, LIST-of-MAPs
- MAP assertions use `element_at`/`cardinality` instead of whole-map string rendering (key ordering is non-deterministic over Arrow Flight)
- INSERT...SELECT (remote→remote and local→remote), UPDATE, DELETE, transactions

**DuckLake conformance** (`ducklake_struct_conformance.test`, `ducklake_map_conformance.test`):
- Document native DuckLake behavior so integration failures can be attributed correctly

**C++ unit tests** (`test_insert_sql_builder.cpp`):
- 18 new STRUCT/MAP tests: basic, NULL, nested, mixed types, SQL injection edge cases
- Single-quote escaping in MAP keys, STRUCT field names, and STRUCT VARCHAR values

**Roadmap**: RM14 moved from current targets to graduated targets; roadmap test file deleted.

## Test results

| Layer | Tests | Assertions |
|-------|-------|------------|
| C++ unit | 347 passed, 2 skipped | 689 |
| Integration | 41 passed | 2874 |
| `make tidy-check` | clean | — |
| `make format-fix` | clean | — |

## Test plan

- [x] `make format-fix` clean
- [x] `make tidy-check` clean
- [x] C++ unit tests pass (347 passed, 2 skipped — preexisting CTE rewriter)
- [x] Full integration suite green (41 tests, 2874 assertions)
- [x] DuckLake conformance tests pass

### CODE REVIEW

File: test/sql/unit/ducklake_list_conformance.test [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+statement ok
+INSERT INTO dl.main.t VALUES (1, [10, 20, 30]);
+
+query IT
+SELECT id, l FROM dl.main.t;
+----
+1	[10, 20, 30]
+
+# ============================================================
+# LIST functions: list_extract, list_length, list_concat
```
[the remark this answers is not in the document]
Comment (jghoman) [reply test/sql/unit/ducklake_list_conformance.test]: Fixed.

File: test/cpp/test_insert_sql_builder.cpp [hunk 2]
```diff
@@ -264,6 +264,239 @@ TEST_CASE("Insert SQL builder - TIMESTAMP", "[duckhog][insert-sql]") {
...
+	auto list_type = LogicalType::LIST(struct_type);
+	DataChunk chunk;
+	InitChunk(chunk, {list_type},
+	          {Value::LIST(struct_type, {Value::STRUCT({{"name", Value("apple")}, {"qty", Value::INTEGER(5)}}),
+	                                     Value::STRUCT({{"name", Value("banana")}, {"qty", Value::INTEGER(3)}})})});
+	auto sql = BuildInsertSQL(TABLE, {"items"}, chunk);
+	REQUIRE(sql == "INSERT INTO ducklake.myschema.t (items) VALUES ([{'name': 'apple', 'qty': 5}, {'name': 'banana', "
+	               "'qty': 3}]);");
+}
+
```
[the remark this answers is not in the document]
Comment (jghoman) [reply test/cpp/test_insert_sql_builder.cpp:467]: Implemented.

File: test/sql_roadmap.md [hunk 3]
```diff
@@ -69,6 +68,7 @@ Graduated targets (now part of normal integration suite):
 - [`time_travel_remote.test_slow`](sql/integration/time_travel_remote.test_slow) (RM10) — [#33](https://github.com/PostHog/duckhog/issues/33)
 - [`table_functions_remote.test_slow`](sql/integration/table_functions_remote.test_slow) (RM11, RM12, RM13) — [#34](https://github.com/PostHog/duckhog/issues/34), [#35](https://github.com/PostHog/duckhog/issues/35), [#36](https://github.com/PostHog/duckhog/issues/36)
 - [`insert_default_values_remote.test_slow`](sql/integration/insert_default_values_remote.test_slow) (RM18)
+- [`struct_types_remote.test_slow`](sql/integration/struct_types_remote.test_slow), [`map_types_remote.test_slow`](sql/integration/map_types_remote.test_slow) (RM14) — [#37](https://github.com/PostHog/duckhog/issues/37) — LIST/STRUCT/MAP nested types; Duckgres Arrow Flight fix + DuckHog `ValueToInsertSQL` serializer
```
[the remark this answers is not in the document]
Comment (jghoman) [reply test/sql_roadmap.md]: Done.
